### PR TITLE
Schedules and students enhancements

### LINKED ISSUE #708: Students have access to all events /api/v1/schedules/events

### LINKED ISSUE #730: Students should be able to see their student groups
Endpoint students/{id}/groups should be available for student, but only if requested id is this student's id.

### PR DESCRIPTION
**Pull request by DmitriyFesak**
Improved student and mentor event filtering interactions and provided students with the ability to see their student groups.

### CODE REVIEW

File: CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs [hunk 1]
```diff
@@ -149,6 +152,21 @@ public async Task<Result<IList<ScheduledEventDTO>>> GetEventsFiltered(ScheduledE
                 return Result<IList<ScheduledEventDTO>>.GetError(ErrorCode.ValidationError, error);
             }
 
+            if (_currentUserService.Role == UserRole.Student)
+            {
+                if (_currentUserService.AccountId != request.StudentAccountID)
+                {
+                    return Result<IList<ScheduledEventDTO>>.GetError(ErrorCode.Unauthorized, "Student can see only his events");
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs]: Please replace with 'Student can see only own events'.

File: CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs [hunk 2]
```diff
@@ -149,6 +152,21 @@ public async Task<Result<IList<ScheduledEventDTO>>> GetEventsFiltered(ScheduledE
...
+                }
+            }
+            else if (_currentUserService.Role == UserRole.Mentor)
+            {
+                if (_currentUserService.EntityId != request.MentorID)
+                {
+                    return Result<IList<ScheduledEventDTO>>.GetError(ErrorCode.Unauthorized, "Mentor can see only his events");
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs]: Please replace with 'Mentor can see only own events'

File: CharlieBackend.Business/Services/StudentService.cs [hunk 3]
```diff
@@ -112,6 +115,14 @@ private async Task<IList<StudentDetailsDto>> GetStudentsWithAvatarIncluded(IList
 
         public async Task<Result<IList<StudentStudyGroupsDto>>> GetStudentStudyGroupsByStudentIdAsync(long id)
         {
+            if (_currentUserService.Role == UserRole.Student)
+            {
+                if (_currentUserService.EntityId != id)
+                {
+                    return Result<IList<StudentStudyGroupsDto>>.GetError(ErrorCode.Unauthorized, "Student can see only his student groups");
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/StudentService.cs]: '.... only own student groups'

**Review by andrii-developer — APPROVED** [review]
Please update error messages before merging. Thanks.

**Review by maksymiu4e — APPROVED** [review]

**Review by elizabeth-grant-20 — APPROVED** [review]

**Review by KSudak — APPROVED** [review]

**Review by MakhortovSasha — APPROVED** [review]

File: CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs [hunk 4]
```diff
@@ -149,6 +152,21 @@ public async Task<Result<IList<ScheduledEventDTO>>> GetEventsFiltered(ScheduledE
                 return Result<IList<ScheduledEventDTO>>.GetError(ErrorCode.ValidationError, error);
             }
 
+            if (_currentUserService.Role == UserRole.Student)
```
Comment (andrii-developer) [inline CharlieBackend.Business/Services/ScheduleServiceFolder/ScheduleService.cs:155]: Please merge nested 'if' into single condition. ... && ...

**Pull request merged**